modules/shared/src/main/scala/org/tessellation/currency/dataApplication/package.scala

```diff
@@ -27,7 +35,32 @@ import org.http4s._
 import org.http4s.circe.jsonEncoderOf
 import org.http4s.server.Router
 
-trait DataUpdate
+sealed trait DataTransaction
+object DataTransaction {
+  implicit def encoder(implicit e: Encoder[DataUpdate]): Encoder[DataTransaction] =
+    Encoder.instance {
+      case v: FeeTransaction => v.asJson
+      case v: DataUpdate     => e.apply(v)
+    }
+
+  implicit def decoder(implicit d: Decoder[DataUpdate]): Decoder[DataTransaction] =
+    (c: HCursor) => Decoder[FeeTransaction].apply(c).orElse(d.apply(c))
+}
+
+trait DataUpdate extends DataTransaction
+
+@derive(decoder, encoder, order, ordering, show)
+case class FeeTransaction(
+  source: Address,
+  destination: Address,
+  amount: Amount,
+  dataUpdateRef: Hash
+) extends DataTransaction
+
+object FeeTransaction {
+  def serializeFeeTransaction[F[_]: Async](feeTransaction: FeeTransaction): F[Array[Byte]] =
```

Comment:
this function is already within `FeeTransaction` scope, so there is no need to repeat the name. So just, `FeeTransaction.serialize` instead of `FeeTransaction.serializeFeeTransaction`.

Reply:
👍🏼